### ISSUE TITLE
GradientBoostingSurvivalAnalysis uses only a single CPU core

### ISSUE THREAD

**Issue by KOLANICH** [sebp/scikit-survival#18]
At least on a 2-core cpu its utilization is only 50% when fitting.

It seems that there is a bottleneck in python code, since xgboost utilizes all the cores and that the implementation using xgboost tree regressors is even slower than the one using sklearn.

**Comment by sebp** [comment]
The actual heavy lifting is done by scikit-learn's implementation of gradient boosting, which currently does not support parallelization. At the moment, I don't have plans to support xgboost.
